src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,32 +8,52 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+	public static final String EXAMPLE = "123, Abc Street, #01-55, 123456";
+	public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must follow this format a/BLOCK, STREET, UNIT, POSTAL_CODE";	      
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String ADDRESS_SPLIT_KEYWORD = ", ";
 
     public final String value;
     private boolean isPrivate;
-
+    
+    private static final int ADDRESS_ARRAY_BLOCK = 0;
+    private static final int ADDRESS_ARRAY_STREET = 1;
+    private static final int ADDRESS_ARRAY_UNIT = 2;
+    private static final int ADDRESS_ARRAY_POSTALCODE = 3;
+    
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+    	this.isPrivate = isPrivate;
+    	
+    	String[] args = address.trim().split(ADDRESS_SPLIT_KEYWORD);    	
+    	Block block = new Block(args[ADDRESS_ARRAY_BLOCK]);
+    	Street street = new Street(args[ADDRESS_ARRAY_STREET]);
+    	Unit unit = new Unit(args[ADDRESS_ARRAY_UNIT]);
+    	PostalCode postalCode = new PostalCode(args[ADDRESS_ARRAY_POSTALCODE]);
+        
+        if (!isValidAddress(block, street, unit, postalCode)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        
+        this.value = convertAddressToFriendlyStringFormat(block, street, unit, postalCode);
     }
 
+	private String convertAddressToFriendlyStringFormat(Block block, Street street, Unit unit, PostalCode postalCode) {
+		return block.getValue() + ADDRESS_SPLIT_KEYWORD + street.getValue() + ADDRESS_SPLIT_KEYWORD + unit.getValue() + ADDRESS_SPLIT_KEYWORD + postalCode.getValue();
+	}
+
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if all parts of the address is valid
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public static boolean isValidAddress(Block block, Street street, Unit unit, PostalCode postalCode) {
+    	return  block.getValue().matches(ADDRESS_VALIDATION_REGEX) &&
```

Comment:
Strange that although you kept the original address in your class, you did not use it for this validation.